### PR TITLE
solution

### PR DESCRIPTION
**Pull request by GoldenSpade**
[DEMO LINK](https://BilliZane.github.io/react_list-of-goods/)

### CODE REVIEW

**Review by stas-dukach — APPROVED** [review]
Otlichno!

File: src/ListOFGoods.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import PropTypes from 'prop-types';
+
+export class ListOfGoods extends React.Component {
+  state = {
+    goods: [...this.props.goods],
+  }
+
+  reverseList = () => {
+    this.setState(state => ({
+      goods: state.goods.reverse(),
```
Comment (varseniuk) [inline src/ListOFGoods.js:11]: Don't mutate the state. Create a copy of an array and make all sortings with it

**Review by varseniuk — CHANGES_REQUESTED** [review]
The logic is correct, but you are mutating the state. Create a copy of an array and make all changes with it

File: src/GoodsList.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+class GoodsList extends React.Component {
+  state = {
+    goodsList: [...this.props.goodsList],
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      goodsList: state.goodsList.reverse(),
+    }));
+  }
```
Comment (varseniuk) [inline src/GoodsList.jsx:12]:
```suggestion
  reverse = () => {
    this.setState(state => ({
      goodsList: [...state.goodsList].reverse(),
    }));
  }
```
You're still mutating the state. 'reverse` and `sort` are mutating methods. Create a copy and make all operations with it

File: src/GoodsList.jsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  alphaSort = () => {
+    this.setState(state => ({
+      goodsList: state.goodsList.sort(),
```
Comment (varseniuk) [inline src/GoodsList.jsx:16]:
```suggestion
      goodsList: [...state.goodsList].sort(),
```

File: src/GoodsList.jsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    }));
+  }
+
+  reset = () => {
+    this.setState({ goodsList: [...this.props.goodsList] });
+  }
+
+  lengthSort = () => {
+    this.setState(state => ({
+      goodsList: state.goodsList.sort((a, b) => a.length - b.length),
```
Comment (varseniuk) [inline src/GoodsList.jsx:26]:
```suggestion
      goodsList: [...state.goodsList].sort((a, b) => a.length - b.length),
```

**Review by varseniuk — CHANGES_REQUESTED** [review]
You are still mutating the state

**Review by varseniuk — APPROVED** [review]

**Pull request closed without merging**